src/components/ViewPackagePage/components/MarkdownShikiCodeViewer.tsx

```diff
@@ -0,0 +1,64 @@
+import { useEffect, useMemo } from "react"
+import { createHighlighter, Highlighter } from "shiki"
+
+let globalHighlighter$: any
+let globalHighlighter: Highlighter
+
+const fileExtensionsToLanguages: Record<string, string> = {
+  ts: "typescript",
+  tsx: "typescript",
+}
+
+export const MarkdownShikiCodeViewer = ({
+  code,
+  language,
+}: {
+  code: string
+  language: string
+}) => {
+  useEffect(() => {
+    async function setupHighlighter() {
+      if (globalHighlighter$) return
+      globalHighlighter$ = await createHighlighter({
+        langs: ["typescript"],
+        themes: ["vitesse-light"],
+      })
+      globalHighlighter = await globalHighlighter$
+    }
+    setupHighlighter()
+  }, [])
+
+  const html = useMemo(
+    () =>
+      globalHighlighter?.codeToHtml(code, {
+        lang: fileExtensionsToLanguages[language] || "typescript",
+        theme: "vitesse-light",
+      }),
+    [language, code, globalHighlighter],
+  )
+
+  if (!html) {
+    return <div>Loading...</div>
+  }
+
+  let processedHtml = html
+    .replace(/<span class="line-number">.*?<\/span>/g, "")
+    .replace(/<div class="line-number">.*?<\/div>/g, "")
+    .replace(/data-line="[^"]*"/g, "")
+    .replace(
+      /style="background-color:(.*?)"/g,
+      'style="background-color:#f3f4f6"',
+    )
+
+  processedHtml = processedHtml.replace(
+    /class="shiki/g,
+    'class="shiki markdown-shiki-code',
+  )
```

Comment:
can you find another way to do this? This seems very hacky and prone to breakage